ats/pql/pql.py

```diff
@@ -0,0 +1,186 @@
+
+        name = current_token
+        current_token = get_next_token()
+
+        return name
+
+    def match_variable_in_query_token(variables):
+        assert_token("VARIABLE_IN_QUERY_TOKEN")
+        nonlocal current_token
+        # w przyszłości jak zostanie dodane "and" do pql trzeba będzie to rozszerzyć o ilość zmiennych występujących w zapytaniu, 
+        # póki co są tylko 2 więc narazie to wystarczy
+        if current_token != variables[0] and current_token != variables[1]:
+            raise ValueError(f"Token '{current_token}' is not a valid VARIABLE_IN_QUERY_TOKEN")
+
+        current_token = get_next_token()
+
+
+    def match_design_entity_relationship():
+        assert_token("DESIGN_ENTITY_RELATIONSHIP_TOKEN")
+        nonlocal current_token
+        if not is_program_design_entity_relationship_token(current_token):
+            raise ValueError(f"Token '{current_token}' is not a valid NAME_TOKEN")
+
+        relationship = current_token
+        current_token = get_next_token()
+        if current_token == "*": 
+            relationship += "*"
+            current_token = get_next_token()
+
+        return relationship
+
+
+    def match_end_of_declaration_token():
+        assert_token("NAME_DECLARATION_TOKEN")
+        nonlocal current_token
+        if current_token != "," and current_token != ";":
+            raise ValueError(f"Token '{current_token}' is not a valid NAME_DECLARATION_TOKEN")
+
+        name = current_token
+        current_token = get_next_token()
+
+        return name
+
+    def match_parameter_token(variables):
+        assert_token("RELATIONSHIP_PARAMETER_TOKEN")
+        nonlocal current_token
+        if not is_string_token(current_token) and not is_integer_token(current_token) and variables[current_token] is None:
+            raise ValueError(f"Token '{current_token}' is not a valid RELATIONSHIP_PARAMETER_TOKEN")
+            
+        parameter = current_token
+        current_token = get_next_token()
+
+        return parameter
+
+    def process_pql_code():
+        nonlocal current_token
+        current_token = get_next_token()
+        variables = process_variable({})
+        process_operation(variables)       
+
+        return None
+
+    def process_operation(variables):
+        nonlocal current_token
+        while is_variable_type_token(current_token):
+            variables = process_variable(variables)
+
+        process_query(variables)
+
+        return None
+
+    def process_query(variables):
+        nonlocal current_token
+        match_token("Select")
+        searching_variable = match_variable_is_in_list_token(variables)
+        # is_with = False
+
+        match_token("such")
+        match_token("that")
+        relationship = match_design_entity_relationship()
+        match_token("(")
+        first_parameter = match_parameter_token(variables)
+        match_token(",")
+        second_parameter = match_parameter_token(variables)
+        match_token(")")
```

Comment:
tu i tak bedziesz to musial rozbic bardziej na `process_follows`, `process_parent` itd.